tests/unit/components/playback/controls/PlateBarcode.spec.js

```diff
@@ -57,176 +57,37 @@ describe("PlateBarcode.vue", () => {
-    ],
-    [
-      "*#200440012",
-      "validate_plate_barcode",
-      "error not matching MA MB MD",
-      null,
-    ],
-    ["MA200000012", "validate_plate_barcode", "error as day is 0", null],
-    ["MA203670012", "validate_plate_barcode", "error as day is 367", null],
-    ["MA209990121", "validate_plate_barcode", "error as day is 999", null],
-    ["MA 13000012", "validate_plate_barcode", "error due to <space>", null],
-    ["MA  300000", "validate_plate_barcode", "error due to 2<space>", null],
-    [
-      "MB190440991",
-      "validate_plate_barcode",
-      "year 19 now allowed",
-      "MB190440991",
-    ],
-    [
-      "MB210440991",
-      "validate_plate_barcode",
-      "year 21 now allowed",
-      "MB210440991",
-    ],
-    [
-      "MB100440991",
-      "validate_plate_barcode",
-      "year 10 now allowed",
-      "MB100440991",
-    ],
-    ["MA*#300001", "validate_plate_barcode", "error as *# asterisk", null],
-    ["MA20222111*", "validate_plate_barcode", "error as * asterisk", null],
-    ["MA20010*#12", "validate_plate_barcode", "error as *# asterisk", null],
-    ["MA20001 021", "validate_plate_barcode", "error due <space> in day", null],
-    ["MA20001º21", "validate_plate_barcode", "error due to symbol º", null],
-    ["MA20210न21", "validate_plate_barcode", "error due to unicode", null],
-    ["MA20011浩211", "validate_plate_barcode", "error due to unicode", null],
-    [
-      "MA二千万一千〇九",
-      "validate_plate_barcode",
-      "error due to all unicode",
-      null,
-    ],
-    [
-      "MA",
-      "validate_plate_barcode",
-      "error due to not matching length (10,11)",
-      null,
-    ],
-    [
-      "MA20",
-      "validate_plate_barcode",
-      "error due to not matching length (10,11)",
-      null,
-    ],
-    [
-      "MA20044",
-      "validate_plate_barcode",
-      "error due to not matching length (10,11)",
-      null,
-    ],
-    [
-      "MA20**#*",
-      "validate_plate_barcode",
-      "error due to not matching length (10,11)",
-      null,
-    ],
-    [
-      "MA20044001",
-      "validate_plate_barcode",
-      "All criteria matches",
-      "MA20044001",
-    ],
-    ["M120044099", "validate_plate_barcode", "error as M1 is disallowed", null],
-    [
-      "ME20044099",
-      "validate_plate_barcode",
-      "All criteria matches",
-      "ME20044099",
-    ], // new rule allow MD
-  ])(
-    "Given a barcode with text  %s, When validation rule %s criteria FAILS for invalid barcode or PASSES due %s for a valid barcode, Then only valid barcode %s is stored in Vuex playback.barcode",
-    async (platecode, validation_rule, reason, store_data) => {
-      const spied_text_validator = jest.spyOn(
-        TextValidation.prototype,
-        validation_rule
-      );
+  test("Given a valid barcode has been into the Vuex, When Playback State is at any state, Then the text of the Barcode Input field should be valid barcode string and Red Box is visible ", async () => {
```

Comment:
I'm not sure what line of your test "When Playback State is at any state" relates to.  Is the trigger/Act/When part of this test just mounting the component?

Reply:
Removed "When Playback State" its now "When system_status is in any state"

Reply:
The "When" section is used to describe something that happens during the test. It is the "Act"/"Trigger" of the test.  There must be a specific line in the test that links to this---for example making an axios request, clicking an element, dispatching an action, etc.

I realize that in English there are two different ways the word "When" can be used.  In testing terminology though, it only refers to a trigger, not a pre-condition.  The "Given" section is where the Arrangement / Pre-conditions of the test are described.  

It looks like for these tests that you are seeing what happens during the process of mounting the component, so "When the component is mounted" would be the trigger.  Anything related to the `system_status` would be something in the setup before the action is triggered (so it would be in the Arrange/Given section of the title).  However, if something can be "in any state", then it probably does not need to be described in the test title as it is not a relevant condition for the test. If you want to explicitly test that something works in multiple different starting states, then those starting states should be individually described (I don't think that's necessary for this feature, just providing you with information in general)

Reply:
Sorry, I don't agree, if we remove the the function the the getter function the file 'playback.spec.js', testcases are failing you can comment out the function 'playback/getters.js' and verify.

Kindly see the log I get when I run the following unit test of playback.spec.js
`
● store/playback › mutations › barcode validation and is_valid_barcode verification › Given a barcode scanned results in value ME20044099, When validation rule FAILS  or PASSES due All criteria matches, Then validation results set is_valid_barcode to true

    expect(received).toBe(expected) // Object.is equality

    Expected: true
    Received: undefined

      221 |         async (platecode, reason, valid) => {
      222 |           store.commit("playback/set_barcode_number", platecode);
    > 223 |           expect(store.getters["playback/is_valid_barcode"]).toBe(valid);
          |                                                              ^
      224 |         }
      225 |       );
      226 |     });

      at tests/unit/store/playback.spec.js:223:62

----------------------------------|---------|----------|---------|---------|------------------------
File                              | % Stmts | % Branch | % Funcs | % Lines | Uncovered Line #s      
----------------------------------|---------|----------|---------|---------|------------------------
All files                         |    50.2 |    34.24 |   45.28 |    49.9 |                        
 js_utils                         |   26.64 |    17.65 |   27.27 |   26.67 |                        
  axios_helpers.js                |     100 |      100 |     100 |     100 |                        
  text_validation.js              |   24.39 |    19.35 |   27.27 |   24.22 | 26,43-57,74,83,132-401 
  waveform_data_formatter.js      |   23.61 |        0 |      20 |   23.94 | 32-249                 
 store                            |   55.56 |      100 |      20 |   55.56 |                        
  ApiService.js                   |   42.86 |      100 |       0 |   42.86 | 11-28                  
  index.js                        |     100 |      100 |     100 |     100 |                        
 store/modules/flask              |   85.19 |    68.18 |      70 |   84.91 |                        
  actions.js                      |      80 |    61.11 |   66.67 |      80 | 28-32,42,78            
  enums.js                        |     100 |      100 |     100 |     100 |                        
  getters.js                      |       0 |      100 |       0 |       0 | 5-8                    
  index.js                        |     100 |      100 |     100 |     100 |                        
  mutations.js                    |     100 |      100 |     100 |     100 |                        
  url_regex.js                    |     100 |      100 |     100 |     100 |                        
 store/modules/playback           |   95.83 |      100 |   86.21 |   95.79 |                        
  actions.js                      |   98.55 |      100 |   91.67 |   98.55 | 40                     
  enums.js                        |     100 |      100 |     100 |     100 |                        
  getters.js                      |      50 |      100 |      50 |      50 | 17-23                  
  index.js                        |     100 |      100 |     100 |     100 |                        
  mutations.js                    |     100 |      100 |     100 |     100 |                        
 store/modules/settings           |   16.67 |      100 |    6.25 |   11.76 |                        
  actions.js                      |       0 |        0 |       0 |       0 |                        
  getters.js                      |       0 |      100 |       0 |       0 | 3-32                   
  index.js                        |     100 |      100 |     100 |     100 |                        
  mutations.js                    |       0 |      100 |       0 |       0 | 3-15                   
 store/modules/twentyfourcontrols |      60 |      100 |   33.33 |      50 |                        
  getters.js                      |       0 |      100 |       0 |       0 | 3                      
  index.js                        |     100 |      100 |     100 |     100 |                        
  mutations.js                    |       0 |      100 |       0 |       0 | 3                      
 store/modules/waveform           |      50 |    66.67 |   33.33 |   49.15 |                        
  actions.js                      |   42.31 |       50 |      50 |      44 | 22-54                  
  getters.js                      |   26.67 |      100 |       0 |   28.57 | 18-48                  
  index.js                        |     100 |      100 |     100 |     100 |                        
  mutations.js                    |   57.14 |      100 |   44.44 |   53.85 | 16-21,34-43            
  url_regex.js                    |     100 |      100 |     100 |     100 |                        
----------------------------------|---------|----------|---------|---------|------------------------
Test Suites: 1 failed, 1 total
Tests:       26 failed, 39 passed, 65 total
Snapshots:   0 total
Time:        3.023 s
Ran all test suites matching /tests\/unit\/store\/playback.spec.js/i.
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! @curi-bio/mantarray-frontend-components@0.1.2 test:unit: `npm run build && jest -c jest.config.js "tests/unit/store/playback.spec.js"`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the @curi-bio/mantarray-frontend-components@0.1.2 test:unit script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/ubuntu/.npm/_logs/2020-12-09T12_57_32_015Z-debug.log
raghu:~/environment/mantarray-frontend-components (plate_barcode_scanner_enhancements) $ `